#! /bin/sh /usr/share/dpatch/dpatch-run
## 30_slave_piddir.dpatch by Niko Tyni <ntyni@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Add an optional --pid-dir option to specify the pid directory
## DP: when running as a slave

@DPATCH@
diff -urNad smokeping-git~/bin/smokeping.dist smokeping-git/bin/smokeping.dist
--- smokeping-git~/bin/smokeping.dist	2008-03-04 23:10:42.000000000 +0200
+++ smokeping-git/bin/smokeping.dist	2008-03-04 23:12:09.000000000 +0200
@@ -67,6 +67,9 @@
 
  --cache-dir=s Directory for temporary data storage when running in slave mode.
 
+ --pid-dir=s Directory for the PID file when running in slave mode.
+             (optional, defaults to --cache-dir)
+
 =head1 DESCRIPTION
 
 The B<smokeping> tool is the commandline part of the SmokePing system. Its
diff -urNad smokeping-git~/lib/Smokeping/Slave.pm smokeping-git/lib/Smokeping/Slave.pm
--- smokeping-git~/lib/Smokeping/Slave.pm	2008-03-04 22:34:48.000000000 +0200
+++ smokeping-git/lib/Smokeping/Slave.pm	2008-03-04 23:13:45.000000000 +0200
@@ -103,7 +103,7 @@
         if ($@){
             warn "WARNING evaluating new config from server failed: $@ --\n$data";
         } elsif (defined $config and ref $config eq 'HASH'){
-            $config->{General}{piddir} = $slave_cfg->{cache_dir};
+            $config->{General}{piddir} = $slave_cfg->{pid_dir};
             Smokeping::do_log("Sent data to Server and got new config in response.");
             return $config;
         }                       
diff -urNad smokeping-git~/lib/Smokeping.pm smokeping-git/lib/Smokeping.pm
--- smokeping-git~/lib/Smokeping.pm	2008-03-04 22:34:48.000000000 +0200
+++ smokeping-git/lib/Smokeping.pm	2008-03-04 23:13:20.000000000 +0200
@@ -3949,7 +3949,8 @@
     GetOptions(\%opt, 'version', 'email', 'man:s','help','logfile=s','static-pages:s', 'debug-daemon',
                       'nosleep', 'makepod:s','debug','restart', 'filter=s', 'nodaemon|nodemon',
                       'config=s', 'check', 'gen-examples', 'reload', 
-                      'master-url=s','cache-dir=s','shared-secret=s','slave-name=s') or pod2usage(2);
+                      'master-url=s','cache-dir=s','shared-secret=s',
+                      'slave-name=s','pid-dir=s') or pod2usage(2);
     if($opt{version})  { print "$VERSION\n"; exit(0) };
     if(exists $opt{man}) {
         if ($opt{man}) {
@@ -3989,6 +3990,7 @@
         $slave_cfg = {
             master_url => $opt{'master-url'},
             cache_dir => $opt{'cache-dir'},
+            pid_dir   => $opt{'pid-dir'} || $opt{'cache-dir'},
             shared_secret => $secret,
             slave_name => $opt{'slave-name'} || hostname(),
         };
